packages/ra-core/src/controller/usePaginationState.ts

```diff
@@ -33,15 +48,32 @@ interface PaginationProps {
  * @param {numper} initialPerPage the initial value per page
  * @returns {PaginationProps} The pagination props
  */
-export default (initialPerPage: number = 25): PaginationProps => {
-    const [page, setPage] = useState(1);
-    const [perPage, setPerPage] = useState(initialPerPage);
-    useEffect(() => setPerPage(initialPerPage), [initialPerPage]);
+export default (
+    initialPagination: { perPage?: number; page?: number } = {}
+): PaginationProps => {
+    const [pagination, setPagination] = useReducer(paginationReducer, {
+        ...defaultPagination,
+        ...initialPagination,
+    });
+    const isFirstRender = useRef(true);
+
+    const setPerPage = useMemo(() => perPage => setPagination({ perPage }), []);
+    const setPage = useMemo(() => page => setPagination({ page }), []);
+
+    useEffect(() => {
+        if (isFirstRender.current) {
+            isFirstRender.current = false;
+            return;
+        }
+        setPerPage(initialPagination.perPage || 25);
+    }, [initialPagination.perPage, setPerPage]);
```

Comment:
just to understand, this is just to support the case when the injected props for `perPage` change, right? Tehn why don't you support the case when the injected props for `page` change, too?

Reply:
I added this for retrocompatibility sake, and think it should be removed.
The intial value of a state should not change, if it need to change, then it means that the state should be moved up in the component chain.